### DISCUSSION TITLE
updateLabelPositions alternative?

### DISCUSSION THREAD

**Discussion by bedretchukdevoz** [clientIO/joint#2831] [Q&A]
### Introduction

In version 3.4, within joint.dia.LinkView, there was the updateLabelPositions method, which we used to detect when the label was moved. We used this method to calculate an anchor line for the label relative to the link it was associated with. In version 4.0, we noticed that this method no longer exists, thus breaking our implementation.

Is there another method or event that could be used to work around this incompatibility?

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
It still exists in `v4`: https://github.com/clientIO/joint/blob/master/packages/joint-core/src/dia/LinkView.mjs#L891-L919

---

Here's an example on how to implement it using the custom presentation attributes: https://codepen.io/jointjs/pen/xbKXOgL

_Note: the labels are currently not updated when the link path changes, so the linkView update logic needs to be rewritten a bit._
